src/makielayout/blocks/menu.jl

```diff
@@ -202,41 +179,58 @@ function initialize_block!(m::Menu; default = nothing)
-        i = pick_entry(me)
-        optionpolycolors[] = map(eachindex(optionstrings[])) do j
-            j == m.i_selected[] ? m.cell_color_active[] :
-            i == j ? m.cell_color_hover[] :
-                iseven(i) ? to_color(m.cell_color_inactive_even[]) :
-                to_color(m.cell_color_inactive_odd[]) 
-        end
-        return Consume(false)
-    end
-
-    onmouseout(mouseevents) do me
-        optionpolycolors[] = map(eachindex(optionstrings[])) do i
-            i == m.i_selected[] ? m.cell_color_active[] :
-            iseven(i) ? to_color(m.cell_color_inactive_even[]) :
-                to_color(m.cell_color_inactive_odd[]) 
-        end
-        return Consume(false)
-    end
-
-    onmouseleftclick(mouseevents) do me
-        i = pick_entry(me)
-        m.i_selected[] = i
-        m.is_open[] = false
-        return Consume(true)
-    end
-
-    # To stop other things from triggering on left down inside the dropdown menu
-    onmouseleftdown(_ -> Consume(true), mouseevents)
-
-    # close the menu if the user clicks somewhere else
-    # the logic is a bit convoluted because menuscene and selection poly have to be checked at the same time
-    on(blockscene.events.mousebutton, priority=61) do butt
-        if butt.action === Mouse.release && !is_mouseinside(menuscene) && !(mouseposition_px(blockscene) in selectionarea[])
-            m.is_open[] = false
+    was_inside_options = false
+    was_inside_button = false
+    e = menuscene.events
+    onany(e.mouseposition, e.mousebutton, priority=typemax(Int)) do position, butt
+        mp = screen_relative(menuscene, position)
+        if Makie.is_mouseinside(menuscene)
+            is_over = mouseover(menuscene, optionpolys, optiontexts)
+            pressed = ispressed(menuscene, Mouse.left)
+            if is_over
+                was_inside_options = true
+                if pressed
+                    i = pick_entry(mp[2])
+                    m.i_selected[] = i
+                    m.is_open[] = false
+                    return Consume(true)
+                else
+                    i = pick_entry(mp[2])
+                    optionpolycolors[] = map(eachindex(optionstrings[])) do j
+                        j == m.i_selected[] ? m.cell_color_active[] :
+                        i == j ? m.cell_color_hover[] :
+                            iseven(i) ? to_color(m.cell_color_inactive_even[]) :
+                            to_color(m.cell_color_inactive_odd[])
+                    end
+                end
+            elseif was_inside_options
+                was_inside_options = false
+                optionpolycolors[] = map(eachindex(optionstrings[])) do i
+                    i == m.i_selected[] ? m.cell_color_active[] :
+                    iseven(i) ? to_color(m.cell_color_inactive_even[]) :
+                        to_color(m.cell_color_inactive_odd[])
+                end
+            end
+        elseif Makie.is_mouseinside(blockscene)
```

Comment:
This can just be `else` since `is_mouseinside` just checks area and `blockscene` covers the full window.

Reply:
Actually maybe not since the mouse can trigger outside the window as well... Nvm

Reply:
Hmm, maybe it should trigger outside as well. If you move the mouse quickly you might otherwise suppress the `was_inside_button` true -> false switch.